src/CallableHandler.php

```diff
@@ -122,4 +75,13 @@ public static function execute($callable, array $arguments = [])
             throw $exception;
         }
     }
+
+    protected static function getResolver()
```

Comment:
Ok, so I don't get what is the purpose of the `getResolver()` function.
I'd remove all resolve-related stuff from the CallableHandler to asign just the responsability of invoke callables.

I agree with you about instantiate the resolver in the middleware, but this could be done in the `process` method, to avoid create unnecesary instances. For example, for fast-route could be something like this:

```php
$resolver = $this->container ? new ContainerResolver($this->container) : new ReflectionResolver();
$callable = $resolver->resolve($route[1]);
return Utils\CallableHandler::execute($callable, $arguments);
```